pkg/executor/wasm/handler.go

```diff
@@ -0,0 +1,189 @@
+	// TODO we have been ignoring errors from this method for ages. Now that we actually check them tests fail! nice..
+	// v1.0.3: https://github.com/bacalhau-project/bacalhau/blob/v1.0.3/pkg/executor/wasm/executor.go#L243
+	// current: https://github.com/bacalhau-project/bacalhau/blob/ff1bd9cb1c09fa3652c4a68943a97476340dbe33/pkg/executor/wasm/executor.go#L216
+	for _, importModule := range h.arguments.ImportModules {
+		_, err := loader.InstantiateRemoteModule(ctx, importModule)
+		if err != nil {
+			h.logger.Warn().
+				Str("input_source", importModule.InputSource.Source.Type).
+				Str("input_alias", importModule.InputSource.Alias).
+				Str("input_target", importModule.InputSource.Target).
+				Str("volume_type", importModule.Volume.Type.String()).
+				Str("volume_source", importModule.Volume.Source).
+				Str("volume_target", importModule.Volume.Target).
+				Msg("failed to instantiate import module")
+			// lets just ignore the error like we have always done!
+			h.result = executor.NewFailedResult(
+				fmt.Errorf("failed to instantiate import module (%s): %w",
+					importModule.InputSource.Source.Type, err).Error())
+			return
+		}
+	}
+
+	// Load and instantiate the entry module.
+	entryModule := h.arguments.EntryModule
+	instance, err := loader.InstantiateRemoteModule(ctx, entryModule)
+	if err != nil {
+		h.logger.Warn().
+			Str("input_source", entryModule.InputSource.Source.Type).
+			Str("input_alias", entryModule.InputSource.Alias).
+			Str("input_target", entryModule.InputSource.Target).
+			Str("volume_type", entryModule.Volume.Type.String()).
+			Str("volume_source", entryModule.Volume.Source).
+			Str("volume_target", entryModule.Volume.Target).
+			Msg("failed to instantiate entry module")
+		h.result = executor.NewFailedResult(
+			fmt.Errorf("failed to instantiate entry module module (%s): %w",
+				entryModule.InputSource.Source.Type, err).Error())
+		return
+	}
+
+	// invoke the function entry point
+	entryFunc := instance.ExportedFunction(h.arguments.EntryPoint)
+	h.logger.Info().Msg("running execution")
+
+	// TODO this is a little racey
+	h.running.Store(true)
+	close(h.activeCh)
+
+	// The function should exit which results in a sys.ExitError. So we capture
+	// the exit code for inclusion in the job output, and ignore the return code
+	// from the function (most WASI compilers will not give one). Some compilers
+	// though do not set an exit code, so we use a default of -1.
+	_, wasmErr := entryFunc.Call(wasmCtx)
+	exitCode := int64(0)
+	var errExit *sys.ExitError
+	if errors.As(wasmErr, &errExit) {
+		exitCode = int64(errExit.ExitCode())
+		wasmErr = nil
+		h.logger.Info().Int64("exit_code", exitCode).Msg("execution ended")
+	}
+	if wasmErr != nil {
+		// in the event that an error is returned without an exist code we'll assume the operation
+		// failed and set the exit code to 1
+		exitCode = 1
+		h.logger.Warn().Int64("exit_code", exitCode).Err(wasmErr).Msg("execution ended")
+	}
```

Comment:
This a change from the original code which I don't think is necessary or explained anywhere. As explained in the comment above, the default exit code should be -1 unless one is supplied by the WASM runtime. There is no requirement to return any specific exit code in the case of an error.

Please revert this and we can discuss elsewhere.

Reply:
Sorry, this is my bad. -1 seemed like an odd number to return for an error code, so I asked that it be changed to something more traditional.